test/coinselector.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { Coin } from '../src/wallet/coin';
+import { CoinSelector } from '../src/wallet/coin-selector.ts';
+import { Transaction } from 'bitcoinjs-lib';
+
+describe('CoinSelector', () => {
+    let coinSelector: CoinSelector;
+    const feeRate: number = 5;
+
+    beforeAll(() => {
+        coinSelector = new CoinSelector(feeRate);
+    });
+
+    it('should generate correct cost of change', () => {
+        const outputSize = 31;
+        const inputSizeOfChangeUTXO = 68;
+
+        // Calculate cost of change based on output size and input size
+        const costOfChangeOutput = outputSize * feeRate;
+        const costOfSpendingChange =
+            inputSizeOfChangeUTXO * CoinSelector.LONG_TERM_FEERATE;
+        const correctCostOfChange = costOfChangeOutput + costOfSpendingChange;
+
+        // Get the calculated cost of change from the CoinSelector
+        const costOfChange = coinSelector.costOfChange;
+
+        // Assert the correctness of the CoinSelector result
+        expect(correctCostOfChange).toBe(costOfChange);
+    });
+
+    it('should add change to fee if it is dust', () => {
+        const coins: Coin[] = [];
+        let totalBalance: number = 0;
+
+        // Generate random coins and calculate total balance
+        for (let i = 0; i < 20; i++) {
+            const value = Math.random() * 1000;
+            const coin = new Coin({ value });
+            coins.push(coin);
+            // Calculate total balance by subtracting spending fee
+            totalBalance += coin.value - coin.estimateSpendingFee(feeRate);
+        }
+
+        // Create a new transaction
+        const transaction: Transaction = new Transaction();
+        // Add an output with random scriptPubKey, we only care about its value
+        const randomString = [...Array(50)]
+            .map(() => Math.random().toString(36)[2])
+            .join('');
+        transaction.addOutput(Buffer.from(randomString, 'hex'), 1);
```

Comment:
Can we not have randomness and constant list of strings?

Reply:
Yes, I agree. Though this is just a place holder, the output script has no impact on the coin-selector file.